lib/lolcommits/configuration.rb

```diff
@@ -23,8 +23,15 @@ def configuration_file
 
     def loldir
       return @loldir if @loldir
-      basename ||= File.basename(Git.open('.').dir.to_s).sub(/^\./, 'dot')
-      basename.sub!(/ /, '-')
+      if GitInfo.repo_root?('.')
+        basename ||= File.basename(Git.open('.').dir.to_s).sub(/^\./, 'dot')
```

Comment:
I'm wondering if this basename logic could perhaps be encapsulated into a common method for VcsInfo subclasses, so we can separate the concern of knowing all possibilities in the Configuration class.